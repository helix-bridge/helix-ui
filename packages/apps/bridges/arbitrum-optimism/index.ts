export * from './Arbitrum2Optimism';
export * from './Optimism2Arbitrum';
