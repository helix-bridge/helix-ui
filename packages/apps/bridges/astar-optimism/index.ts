export * from './Astar2Optimism';
export * from './Optimism2Astar';
