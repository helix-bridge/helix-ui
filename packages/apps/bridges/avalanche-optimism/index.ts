export * from './Avalanche2Optimism';
export * from './Optimism2Avalanche';
