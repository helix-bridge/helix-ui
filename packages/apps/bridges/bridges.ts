import { CHAIN_TYPE } from 'shared/config/env';
import { arbitrumAstar } from './celer/arbitrum-astar/utils/bridge';
import { arbitrumAvalanche } from './celer/arbitrum-avalanche/utils/bridge';
import { arbitrumOptimism } from './celer/arbitrum-optimism/utils/bridge';
import { arbitrumPolygon } from './celer/arbitrum-polygon/utils/bridge';
import { astarAvalanche } from './celer/astar-avalanche/utils/bridge';
import { astarOptimism } from './celer/astar-optimism/utils/bridge';
import { avalancheOptimism } from './celer/avalanche-optimism/utils/bridge';
import { avalanchePolygon } from './celer/avalanche-polygon/utils/bridge';
import { bscArbitrum } from './celer/bsc-arbitrum/utils/bridge';
import { bscAstar } from './celer/bsc-astar/utils/bridge';
import { bscAvalanche } from './celer/bsc-avalanche/utils/bridge';
import { bscOptimism } from './celer/bsc-optimism/utils/bridge';
import { bscPolygon } from './celer/bsc-polygon/utils/bridge';
import { crabDVMAstar } from './celer/crabDVM-astar/utils/bridge';
import { crabDVMEthereum } from './celer/crabDVM-ethereum/utils/bridge';
import { crabDVMHeco } from './celer/crabDVM-heco/utils/bridge';
import { crabDVMPolygon } from './celer/crabDVM-polygon/utils/bridge';
import { ethereumArbitrum } from './celer/ethereum-arbitrum/utils/bridge';
import { ethereumAstar } from './celer/ethereum-astar/utils/bridge';
import { ethereumAvalanche } from './celer/ethereum-avalanche/utils/bridge';
import { ethereumBSC } from './celer/ethereum-bsc/utils/bridge';
import { ethereumHeco } from './celer/ethereum-heco/utils/bridge';
import { ethereumOptimism } from './celer/ethereum-optimism/utils/bridge';
import { ethereumPolygon } from './celer/ethereum-polygon/utils/bridge';
import { hecoPolygon } from './celer/heco-polygon/utils/bridge';
import { polygonAstar } from './celer/polygon-astar/utils/bridge';
import { polygonOptimism } from './celer/polygon-optimism/utils/bridge';
import { crabCrabDVM, darwiniaDarwiniaDVM, pangolinPangolinDVM } from './helix/substrate-dvm/utils/bridge';
import { darwiniaCrabDVM, pangoroPangolinDVM } from './helix/substrate-substrateDVM/utils/bridge';
import { crabCrabParachain, pangolinPangolinParachain } from './helix/substrate-substrateParachain/utils/bridge';
import { pangoroDVMGoerli, darwiniaDVMEthereum } from './helix/substrateDVM-ethereum/utils/bridge';
import { darwiniaDVMCrabDVM, pangoroDVMPangolinDVM } from './helix/substrateDVM-substrateDVM/utils/bridge';
import { darwiniaDVMDarwiniaDVM, pangoroDVMPangoroDVM } from './helix/substrateDVM-substrateDVM/utils/bridge-inner';
import { crabParachainKarura } from './xcm/crabParachain-karura/utils/bridge';
import { crabParachainMoonriver } from './xcm/crabParachain-moonriver/utils/bridge';

const formalBridges = [
  arbitrumAstar,
  arbitrumAvalanche,
  arbitrumOptimism,
  astarAvalanche,
  astarOptimism,
  avalancheOptimism,
  bscArbitrum,
  bscAstar,
  bscAvalanche,
  bscOptimism,
  crabCrabDVM,
  crabDVMEthereum,
  crabDVMHeco,
  crabDVMPolygon,
  darwiniaDarwiniaDVM,
  darwiniaCrabDVM,
  darwiniaDVMCrabDVM,
  darwiniaDVMDarwiniaDVM,
  darwiniaDVMEthereum,
  ethereumHeco,
  ethereumPolygon,
  hecoPolygon,
  crabCrabParachain,
  polygonAstar,
  ethereumAstar,
  ethereumBSC,
  ethereumAvalanche,
  ethereumArbitrum,
  ethereumOptimism,
  bscPolygon,
  avalanchePolygon,
  arbitrumPolygon,
  polygonOptimism,
  crabDVMAstar,
  crabParachainKarura,
  crabParachainMoonriver,
];

const testBridges = [
  pangolinPangolinDVM,
  pangoroDVMPangolinDVM,
  pangoroPangolinDVM,
  pangolinPangolinParachain,
  pangoroDVMGoerli,
  pangoroDVMPangoroDVM,
];

export const BRIDGES = (() => {
  switch (CHAIN_TYPE) {
    case 'all':
      return [...formalBridges, ...testBridges];
    case 'formal':
      return formalBridges;
    case 'test':
      return testBridges;
    default:
      return [];
  }
})();
