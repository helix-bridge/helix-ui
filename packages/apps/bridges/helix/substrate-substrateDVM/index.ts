export * from './Substrate2SubstrateDVM';
export * from './SubstrateDVM2Substrate';
