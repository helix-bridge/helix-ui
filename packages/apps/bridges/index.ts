export { Arbitrum2Astar, Astar2Arbitrum } from './cbridge/arbitrum-astar';
export { Arbitrum2Avalanche, Avalanche2Arbitrum } from './cbridge/arbitrum-avalanche';
export { Arbitrum2Optimism, Optimism2Arbitrum } from './cbridge/arbitrum-optimism';
export { Arbitrum2Polygon, Polygon2Arbitrum } from './cbridge/arbitrum-polygon';
export { Astar2Avalanche, Avalanche2Astar } from './cbridge/astar-avalanche';
export { Astar2Optimism, Optimism2Astar } from './cbridge/astar-optimism';
export { Avalanche2Optimism, Optimism2Avalanche } from './cbridge/avalanche-optimism';
export { Avalanche2Polygon, Polygon2Avalanche } from './cbridge/avalanche-polygon';
export { Arbitrum2BSC, BSC2Arbitrum } from './cbridge/bsc-arbitrum';
export { Astar2BSC, BSC2Astar } from './cbridge/bsc-astar';
export { Avalanche2BSC, BSC2Avalanche } from './cbridge/bsc-avalanche';
export { BSC2Optimism, Optimism2BSC } from './cbridge/bsc-optimism';
export { BSC2Polygon, Polygon2BSC } from './cbridge/bsc-polygon';
export { Astar2CrabDVM, CrabDVM2Astar } from './cbridge/crabDVM-astar';
export { CrabDVM2Ethereum, Ethereum2CrabDVM } from './cbridge/crabDVM-ethereum';
export { CrabDVM2Heco, Heco2CrabDVM } from './cbridge/crabDVM-heco';
export { CrabDVM2Polygon, Polygon2CrabDVM } from './cbridge/crabDVM-polygon';
export { Arbitrum2Ethereum, Ethereum2Arbitrum } from './cbridge/ethereum-arbitrum';
export { Astar2Ethereum, Ethereum2Astar } from './cbridge/ethereum-astar';
export { Avalanche2Ethereum, Ethereum2Avalanche } from './cbridge/ethereum-avalanche';
export { BSC2Ethereum, Ethereum2BSC } from './cbridge/ethereum-bsc';
export { Ethereum2Heco, Heco2Ethereum } from './cbridge/ethereum-heco';
export { Ethereum2Optimism, Optimism2Ethereum } from './cbridge/ethereum-optimism';
export { Ethereum2Polygon, Polygon2Ethereum } from './cbridge/ethereum-polygon';
export { Heco2Polygon, Polygon2Heco } from './cbridge/heco-polygon';
export { Astar2Polygon, Polygon2Astar } from './cbridge/polygon-astar';
export { Optimism2Polygon, Polygon2Optimism } from './cbridge/polygon-optimism';
export { DVM2Substrate, Substrate2DVM } from './helix/substrate-dvm';
export { SubstrateParachain2Substrate, Substrate2SubstrateParachain } from './helix/substrate-substrateParachain';
export { Substrate2SubstrateDVM, SubstrateDVM2Substrate } from './helix/substrate-substrateDVM';
export { Ethereum2SubstrateDVM, SubstrateDVM2Ethereum } from './helix/substrateDVM-ethereum';
export { DarwiniaDVM2CrabDVM, CrabDVM2DarwiniaDVM, DarwiniaDVMInner } from './helix/darwiniaDVM-crabDVM';
export { Unavailable2Unknown, Unknown2Unavailable } from './unknown-unavailable';
export { CrabParachain2Karura, Karura2CrabParachain } from './xcm/crabParachain-karura';
export { CrabParachain2Moonriver, Moonriver2CrabParachain } from './xcm/crabParachain-moonriver';
export { CrabDVMInner } from './helix/crabDVM-darwiniaDVM';

export {
  SubstrateDVM2SubstrateParachain,
  SubstrateParachain2SubstrateDVM,
} from './helix/substrateDVM-substrateParachain';
