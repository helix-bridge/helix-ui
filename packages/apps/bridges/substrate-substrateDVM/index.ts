export * from './Substrate2SubstrateDVM';
export * from './SubstrateDVM2Substrate';
export * from './Substrate2SubstrateDVMRecord';
export * from './SubstrateDVM2SubstrateRecord';
