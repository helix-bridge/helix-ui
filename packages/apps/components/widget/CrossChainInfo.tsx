import { Form, Spin, Typography } from 'antd';
import { PropsWithChildren } from 'react';
import { Bridge } from 'shared/model';

interface CrossChainInfoProps {
  bridge: Bridge;
  fee: {
    amount: string;
    symbol: string;
  } | null;
}

export function CrossChainInfo({ bridge, fee, children }: PropsWithChildren<CrossChainInfoProps>) {
  return (
    <Form.Item label="Info" className="relative">
      <div className="h-20 w-full flex flex-col justify-center space-y-2 px-4 bg-gray-900">
        <div className="flex justify-between items-center">
          <Typography.Text>Bridge Name</Typography.Text>
          <Typography.Text className="capitalize">{bridge?.category}</Typography.Text>
        </div>

        <div className="flex justify-between items-center">
          <Typography.Text>Transaction Fee</Typography.Text>
          {fee ? (
            <Typography.Text>
              {fee.amount} {fee.symbol}
            </Typography.Text>
          ) : (
            <Spin spinning size="small"></Spin>
          )}
        </div>

        {children}
      </div>
    </Form.Item>
  );
}
