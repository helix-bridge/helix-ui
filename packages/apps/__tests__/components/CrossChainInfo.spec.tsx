/// <reference types="jest" />

import { Typography } from 'antd';
import { create } from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { pangoroPangolinDVM } from 'shared/config/bridges/substrate-substrateDVM';
import { CrossChainInfo } from '../../components/widget/CrossChainInfo';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe('<CrossChainInfo />', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('render cross-chain item', () => {
    const component = create(<CrossChainInfo bridge={pangoroPangolinDVM} fee={{ amount: '8', symbol: 'PRING' }} />);

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('render cross-chain with children', () => {
    const component = create(
      <CrossChainInfo bridge={pangoroPangolinDVM} fee={{ amount: '8', symbol: 'RING' }}>
        <span>can receive children node</span>
      </CrossChainInfo>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('render cross-chain and extra nodes', () => {
    const component = create(
      <CrossChainInfo
        bridge={pangoroPangolinDVM}
        fee={{ amount: '8', symbol: 'RING' }}
        extra={[
          {
            name: 'Allowance',
            content: (
              <Typography.Text className="capitalize">
                <span>10000000</span>
                <span className="capitalize ml-1">PRING</span>
              </Typography.Text>
            ),
          },
          {
            name: 'Daily limit',
            content: (
              <Typography.Text className="capitalize">
                <span>100000000</span>
                <span className="capitalize ml-1">PRING</span>
              </Typography.Text>
            ),
          },
        ]}
      ></CrossChainInfo>
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('render cross-chain with dynamic fee', () => {
    const component = create(<CrossChainInfo bridge={pangoroPangolinDVM} isDynamicFee />);

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should toggle extra info on click', () => {
    const { getByTestId } = render(
      <CrossChainInfo
        bridge={pangoroPangolinDVM}
        fee={{ amount: '8', symbol: 'RING' }}
        extra={[
          {
            name: 'Allowance',
            content: (
              <Typography.Text className="capitalize">
                <span>10000000</span>
                <span className="capitalize ml-1">PRING</span>
              </Typography.Text>
            ),
          },
          {
            name: 'Daily limit',
            content: (
              <Typography.Text className="capitalize">
                <span>100000000</span>
                <span className="capitalize ml-1">PRING</span>
              </Typography.Text>
            ),
          },
        ]}
      ></CrossChainInfo>
    );

    expect(getByTestId('Allowance')).toHaveClass('hidden');

    fireEvent.click(getByTestId('toggle-btn'));

    expect(getByTestId('Allowance')).not.toHaveClass('hidden');
  });
});
