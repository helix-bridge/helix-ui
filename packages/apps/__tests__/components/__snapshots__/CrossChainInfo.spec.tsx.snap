// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CrossChainInfo /> render cross-chain and extra nodes 1`] = `
<div
  className="ant-row ant-form-item relative"
  style={Object {}}
>
  <div
    className="ant-col ant-form-item-label"
    style={Object {}}
  >
    <label
      className=""
      title="Information"
    >
      Information
    </label>
  </div>
  <div
    className="ant-col ant-form-item-control"
    style={Object {}}
  >
    <div
      className="ant-form-item-control-input"
    >
      <div
        className="ant-form-item-control-input-content"
      >
        <div
          className="w-full flex flex-col justify-center space-y-2 p-4 bg-gray-900"
        >
          <div
            className="flex justify-between items-center"
          >
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Bridge Name
            </span>
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              helix
            </span>
          </div>
          <div
            className="flex justify-between items-center "
          >
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Transaction Fee
            </span>
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              <span
                className="cursor-help"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                8
              </span>
              <span
                className="ml-1"
              >
                RING
              </span>
            </span>
          </div>
          <div
            className="flex items-center gap-2 cursor-pointer"
            onClick={[Function]}
          >
            <span
              className="flex-1 bg-gray-800 h-px"
            />
            <span
              aria-label="caret-down"
              className="anticon anticon-caret-down text-gray-700 transform transition-all duration-300 "
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="caret-down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="0 0 1024 1024"
                width="1em"
              >
                <path
                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                />
              </svg>
            </span>
            <span
              className="flex-1 bg-gray-800 h-px"
            />
          </div>
          <div
            className="justify-between items-center transition-all duration-100 hidden"
          >
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Allowance
            </span>
            <span
              className="ant-typography capitalize"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              <span>
                10000000
              </span>
              <span
                className="capitalize ml-1"
              >
                PRING
              </span>
            </span>
          </div>
          <div
            className="justify-between items-center transition-all duration-100 hidden"
          >
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Daily limit
            </span>
            <span
              className="ant-typography capitalize"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              <span>
                100000000
              </span>
              <span
                className="capitalize ml-1"
              >
                PRING
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CrossChainInfo /> render cross-chain item 1`] = `
<div
  className="ant-row ant-form-item relative"
  style={Object {}}
>
  <div
    className="ant-col ant-form-item-label"
    style={Object {}}
  >
    <label
      className=""
      title="Information"
    >
      Information
    </label>
  </div>
  <div
    className="ant-col ant-form-item-control"
    style={Object {}}
  >
    <div
      className="ant-form-item-control-input"
    >
      <div
        className="ant-form-item-control-input-content"
      >
        <div
          className="w-full flex flex-col justify-center space-y-2 p-4 bg-gray-900"
        >
          <div
            className="flex justify-between items-center"
          >
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Bridge Name
            </span>
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              helix
            </span>
          </div>
          <div
            className="flex justify-between items-center "
          >
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Transaction Fee
            </span>
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              <span
                className="cursor-help"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                8
              </span>
              <span
                className="ml-1"
              >
                PRING
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CrossChainInfo /> render cross-chain with children 1`] = `
<div
  className="ant-row ant-form-item relative"
  style={Object {}}
>
  <div
    className="ant-col ant-form-item-label"
    style={Object {}}
  >
    <label
      className=""
      title="Information"
    >
      Information
    </label>
  </div>
  <div
    className="ant-col ant-form-item-control"
    style={Object {}}
  >
    <div
      className="ant-form-item-control-input"
    >
      <div
        className="ant-form-item-control-input-content"
      >
        <div
          className="w-full flex flex-col justify-center space-y-2 p-4 bg-gray-900"
        >
          <div
            className="flex justify-between items-center"
          >
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Bridge Name
            </span>
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              helix
            </span>
          </div>
          <div
            className="flex justify-between items-center "
          >
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Transaction Fee
            </span>
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              <span
                className="cursor-help"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                8
              </span>
              <span
                className="ml-1"
              >
                RING
              </span>
            </span>
          </div>
          <span>
            can receive children node
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CrossChainInfo /> render cross-chain with dynamic fee 1`] = `
<div
  className="ant-row ant-form-item relative"
  style={Object {}}
>
  <div
    className="ant-col ant-form-item-label"
    style={Object {}}
  >
    <label
      className=""
      title="Information"
    >
      Information
    </label>
  </div>
  <div
    className="ant-col ant-form-item-control"
    style={Object {}}
  >
    <div
      className="ant-form-item-control-input"
    >
      <div
        className="ant-form-item-control-input-content"
      >
        <div
          className="w-full flex flex-col justify-center space-y-2 p-4 bg-gray-900"
        >
          <div
            className="flex justify-between items-center"
          >
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Bridge Name
            </span>
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              helix
            </span>
          </div>
          <div
            className="flex justify-between items-center "
          >
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              Transaction Fee
            </span>
            <span
              className="ant-typography"
              onClick={null}
              style={
                Object {
                  "WebkitLineClamp": undefined,
                }
              }
            >
              <span
                className="cursor-help"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                8
              </span>
              <span
                className="ml-1"
              >
                RING
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
